terra_notebook_utils/drs.py

```diff
@@ -200,50 +201,62 @@ def _resolve_local_target(filepath: str, info: DRSInfo) -> str:
         filepath = os.path.join(os.path.abspath(filepath), filename)
     return filepath
 
-def copy_to_local(drs_url: str,
-                  filepath: str,
-                  workspace_name: Optional[str]=WORKSPACE_NAME,
-                  workspace_namespace: Optional[str]=WORKSPACE_GOOGLE_PROJECT):
-    """Copy a DRS object to the local filesystem."""
+def _do_copy_drs(drs_uri: str,
+                 dst: str,
+                 multipart_threshold: int,
+                 workspace_name: Optional[str]=WORKSPACE_NAME,
+                 workspace_namespace: Optional[str]=WORKSPACE_GOOGLE_PROJECT):
+    dst_blob: Union[GSBlob, URLBlob, LocalBlob]
+    if dst.startswith("gs://"):
+        src_info = get_drs_info(drs_uri)
+        src_blob = get_drs_blob(src_info, workspace_namespace)
+        bucket_name, key = _resolve_bucket_target(dst, src_info)
+        dst_blob = GSBlob(bucket_name, key)
+        copy_client._do_copy(src_blob, dst_blob, multipart_threshold)
+    else:
+        info = get_drs_info(drs_uri)
+        src_blob = get_drs_blob(info, workspace_namespace)
+        dst_blob = copy_client.blob_for_url(_resolve_local_target(dst, info))
+        copy_client._do_copy(src_blob, dst_blob, multipart_threshold)
+
+class DRSCopyClient(copy_client.CopyClient):
+    workspace: Optional[str] = None
+    workspace_namespace: Optional[str] = None
+
+    def copy(self, drs_uri: str, dst: str):  # type: ignore
+        self._queue.put(_do_copy_drs,
+                        drs_uri,
+                        dst,
+                        self.multipart_threshold,
+                        self.workspace,
+                        self.workspace_namespace)
+
+def copy(drs_uri: str,
+         dst: str,
+         workspace_name: Optional[str]=WORKSPACE_NAME,
+         workspace_namespace: Optional[str]=WORKSPACE_GOOGLE_PROJECT):
+    """Copy a DRS object to either the local filesystem, or to a Google Storage location if `dst` starts with
+    "gs://".
+    """
     enable_requester_pays(workspace_name, workspace_namespace)
-    info = get_drs_info(drs_url)
-    copy_client.copy(get_drs_blob(info, workspace_namespace), _resolve_local_target(filepath, info))
+    with DRSCopyClient(raise_on_error=True, progress_indicator="bar") as cc:
+        cc.workspace = workspace_name
+        cc.workspace_namespace = workspace_namespace
+        cc.copy(drs_uri, dst or ".")
 
-def copy_to_bucket(drs_url: str,
-                   dst_key: str,
+def copy_to_bucket(drs_uri: str,
+                   dst_key: str="",
                    dst_bucket_name: str=None,
```

Comment:
Not part of the PR but `Optional[str]`.

Reply:
An empty string seems fine here.

Reply:
I agree, I meant for the one under it that had a `None` already.